### DISCUSSION TITLE
‏‏‎  Example get storage, and firestore get

### DISCUSSION THREAD

**Discussion by ghost** [SchneiderInfosystems/FB4D#97] [Q&A] [answered]
‏‏my i ask for
example prototype get storage, and firestore get ,
listview as component‏‏‎ ‎

**Comment by SchneiderInfosystems** [comment, marked as answer]
Please check the Firestore Chat Application:

https://github.com/SchneiderInfosystems/FB4D/wiki/Firestore-Chat-Application

https://github.com/SchneiderInfosystems/FB4D/tree/master/Samples/FirestoreSimpleChat

**Reply by SchneiderInfosystems** [reply]
Please read the wiki! 

https://github.com/SchneiderInfosystems/FB4D/wiki/FB4D-Reference-IFirestoreDatabase#firestore-listener

Here you find the answer: **_To be able to monitor all documents in a collection, use SubscribeQuery._**

**Reply by SchneiderInfosystems** [reply]
For the storage, there is no listener at all. 
At first in the write function, you should add the image into the storage before you write the new document for the user into the Firestore. 
So you receive on the listener side the document in the Firestore Listener when the entire content in the document and in the referenced storage is accessible.

**Reply by SchneiderInfosystems** [reply]
In the Firestore Chat Application check e.g. this method: `TfmxChatMain.DownloadProfileImgAndAddToImageList`

**Reply by ghost** [reply]
whether, show accordingly integer des () or random

:)
i I'm a beginner, I don't really understand

**Reply by SchneiderInfosystems** [reply]
Unfortunately, I can no longer help you with the free support. Thank you for your understanding.

**Reply by ghost** [reply]
well thanks, sir christoper
